x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/LookupMergeDropExec.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.physical;
+
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.compute.data.ElementType;
+import org.elasticsearch.xpack.esql.core.expression.Attribute;
+import org.elasticsearch.xpack.esql.core.expression.AttributeSet;
+import org.elasticsearch.xpack.esql.core.expression.FieldAttribute;
+import org.elasticsearch.xpack.esql.core.expression.NamedExpression;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.esql.enrich.AbstractLookupService.LOOKUP_POSITIONS_FIELD;
+
+/**
+ * Physical plan node representing the merge/drop step of a lookup operation.
+ * This handles either merging multiple result pages into one (via MergePositionsOperator)
+ * or dropping the doc block (via ProjectOperator).
+ * This is used internally by lookup services to represent the lookup structure
+ * before converting to operators.
+ */
+public class LookupMergeDropExec extends UnaryExec {
+    private final List<NamedExpression> extractFields;
+    private final ElementType[] mergingTypes;
```

Comment:
Similarly, `ElementType` is generally an operator-level concept - it applies to blocks and loses the distinction between e.g. a keyword column (bytes ref element type) or an ip column (also bytes ref). A list of attributes with proper data types would be nicer.

This may become clear when I go through the rest of the changes, but do the element types and merging channels correspond to columns from the `child` physical plan?

In any case, having a list of attributes with actual data types may be important later to decide pushability of some expressions to Lucene (also happens on the level of physical plans in the rest of the engine). Using a list of attributes also allow us to use the standard query rewriting tooling, like `.forAllExpressions(...)`.